### PR TITLE
impr: Update hearing judge after a hearing (FPLA-3098)

### PR DESCRIPTION
**Pull request by SooryaSu**
### JIRA link (if applicable) ###
[FPLA-3098](https://tools.hmcts.net/jira/browse/FPLA-3098)

### Change description ###
Changes edit journey for the manage hearing event so only the judge details can be edited.
Also prevents a notice of hearing from being sent when a hearing is edited.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageHearingsService.java [hunk 1]
```diff
@@ -112,7 +112,7 @@
         List<Element<HearingBooking>> toBeReListedHearings = caseData.getToBeReListedHearings();
 
         Map<String, Object> listAndLabel = new HashMap<>(Map.of(
-            HEARING_DATE_LIST, asDynamicList(futureHearings),
+            HEARING_LIST, asDynamicList(nonCancelledHearings),
             PAST_HEARING_LIST, asDynamicList(pastAndTodayHearings),
             VACATE_HEARING_LIST, asDynamicList(nonCancelledHearings),
             TO_RE_LIST_HEARING_LIST, asDynamicList(toBeReListedHearings)
```
Comment (SooryaSu) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageHearingsService.java:118]: Need to populate separate lists for `HEARING_DATE_LIST` and `VACATE_HEARING_LIST` for display purposes despite being identical as we cannot mix AND and OR for FieldShowConditions

**Review by jacmurphy — APPROVED** [review]

File: ccd-definition/CaseField/CareSupervision/manageHearingsFields.json [hunk 2]
```diff
@@ -322,34 +322,24 @@
...
   {
     "LiveFrom": "01/01/2017",
     "CaseTypeID": "CARE_SUPERVISION_EPO",
-    "ID": "hasHearingsToVacate",
+    "ID": "hasPastOrFutureHearings",
     "FieldType": "YesOrNo",
     "Label": " ",
     "SecurityClassification": "Public",
-    "Comment": "Used for show/hide message as to when there is no hearings to vacate.",
+    "Comment": "Used for show/hide message as to when there are no hearings to select.",
```
Comment (jacmurphy) [inline ccd-definition/CaseField/CareSupervision/manageHearingsFields.json:342]: thank you for this attention to detail @SooryaSu

**Review by ashwini-mv — APPROVED** [review]

File: ccd-definition/AuthorisationCaseField/CareSupervision/court-admin.json [hunk 3]
```diff
@@ -2291,21 +2298,35 @@
...
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseFieldID": "hasFutureHearings",
     "UserRole": "caseworker-publiclaw-courtadmin",
     "CRUD": "CRU"
   },
   {
     "LiveFrom": "01/01/2017",
     "CaseTypeID": "CARE_SUPERVISION_EPO",
-    "CaseFieldID": "hasFutureHearingDateFlag",
+    "CaseFieldID": "hasHearingsToVacate",
```
Comment (jasonpaige) [inline ccd-definition/AuthorisationCaseField/CareSupervision/court-admin.json:2315]: @SomeshRA is this backward compatible? i think we spoke about this before and it's ok but want to be sure...

File: ccd-definition/AuthorisationCaseField/CareSupervision/court-admin.json [hunk 4]
```diff
@@ -2291,21 +2298,35 @@
...
     "UserRole": "caseworker-publiclaw-courtadmin",
     "CRUD": "CRU"
   },
   {
     "LiveFrom": "01/01/2017",
     "CaseTypeID": "CARE_SUPERVISION_EPO",
-    "CaseFieldID": "noHearingsToEditLabel",
```
Comment (SomeshRA) [inline ccd-definition/AuthorisationCaseField/CareSupervision/court-admin.json:2308]: This needs to be added in deprecated file.

**Pull request merged**